### PR TITLE
add azurelinux as an option for the os type

### LINKED ISSUE #113: [AVM Module Issue]: Azure linux instead of ubuntu
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### (Optional) Module Version

_No response_

### (Optional) Correlation Id

_No response_

### Description

https://learn.microsoft.com/en-us/azure/aks/use-azure-linux

> The Azure Linux container host on AKS uses a native AKS image that provides one place to do all Linux development. Every package is built from source and validated, ensuring your services run on proven components. Azure Linux is lightweight, only including the necessary set of packages needed to run container workloads. It provides a reduced attack surface and eliminates patching and maintenance of unnecessary packages. At the base layer, it has a Microsoft hardened kernel tuned for Azure. Learn more about the [key capabilities of Azure Linux](https://learn.microsoft.com/en-us/azure/azure-linux/intro-azure-linux#azure-linux-container-host-key-benefits).

### PR DESCRIPTION
**Pull request by nellyk**
fixes #113 
closes #113

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -254,7 +253,6 @@ map(object({
     os_disk_size_gb = optional(number, null)
     tags            = optional(map(string), {})
     labels          = optional(map(string), {})
-    zones           = optional(set(string))
```
Comment (nellyk) [inline README.md:257]: i've removed this, based on your comment

File: examples/without_availability_zone/README.md [hunk 2]
```diff
@@ -64,7 +64,7 @@ module "test" {
       orchestrator_version = "1.28"
       max_count            = 110
       min_count            = 2
-      os_sku               = "Ubuntu"
+      os_sku               = "AzureLinux"
```
Comment (nellyk) [inline examples/without_availability_zone/README.md:67]: for testing

File: variables.tf [hunk 3]
```diff
@@ -119,12 +119,11 @@ variable "node_pools" {
...
     min_count       = optional(number)
-    os_sku          = optional(string)
+    os_sku          = optional(string, "AzureLinux")
     mode            = optional(string)
     os_disk_size_gb = optional(number, null)
     tags            = optional(map(string), {})
     labels          = optional(map(string), {})
-    zones           = optional(set(string))
   }))
   default     = {}
```
Comment (zioproto) [inline variables.tf:128]: Question: if the default is `{}` the node pool will be created with Ubuntu that is the AKS default ?

File: examples/without_availability_zone/README.md (hunk 2 above)
Comment (zioproto) [reply examples/without_availability_zone/README.md:67]: What do you mean "for testing" ?

File: variables.tf (hunk 3 above)
Comment (zioproto) [reply variables.tf:128]: I have to test this, but if defaults to Ubuntu we have to change here:

https://github.com/Azure/terraform-azurerm-avm-ptn-aks-production/blob/4a690cb13328006940e2767c309447ddb49129e9/locals.tf#L42C37-L42C43

```
os_sku               = pool.os_sku
```

we can change to something like:

```
os_sku = coalesce(pool.os_sku, "AzureLinux")
```

Comment (zioproto) [reply variables.tf:128]: Forget about the above comments. I tested it and it is working as expected. If you dont specify `os_sku` in the object is going to use as default `AzureLinux` correctly. All good.

**Review by zioproto — APPROVED** [review]

**Review by zioproto — APPROVED** [review]

**Pull request merged**